Order service

LeonKempen opened the pull request.

Nobert1 reviewed (changes requested):
According to the requirements, the order returned by find order has a field called total_price, which I think should be included

LeonKempen commented:
@Nobert1 Good catch! I have updated the order class to also have a total_price. The tests are extended to check if this property is returned and updated properly.

pascalbenschopTU commented:
Is it possible to merge the workflows? Now we do docker compose up twice, which seems a bit unnecessary.

LeonKempen commented:
> Is it possible to merge the workflows? Now we do docker compose up twice, which seems a bit unnecessary.

@pascalbenschopTU Probably, for now the tests do not influence eachother I suppose, so they could be run sequentially. At the moment they run parallel to each other, so it does not really matter that we run docker compose twice

Nobert1 commented on order/app.py:
Could the item list be changed? Right now when we add an item we make a call to get the price of the item, but when we remove the item we make a new call to get the price. I think it's more efficient if items were a dict with the key as the id, and the value as the price of the item. Then we can remove items without making calls to stock. It's also not a problem storing the price since the price of an item can't be updated later.

LeonKempen replied:
I have made the suggested changes.

Nobert1 reviewed (approved):
I really like how you store and read from redis, it's a very clever  way to avoid type mismatching
